### PR TITLE
Fixes milestone in -dependencies bom for ga or rc release.

### PR DESCRIPTION
**Pull request by spencergibb**
Previously if there was a -M* or -RC* in a spring-cloud-*-dependencies/pom.xml this would be skipped. This was do to lines and java regex mismatches. The change moves to using Matcher.lookingAt() which only considers the start of the line, not the end.

Fixes gh-234

### CODE REVIEW

File: releaser-core/src/main/java/releaser/internal/project/ProjectVersion.java [hunk 1]
```diff
@@ -54,7 +54,7 @@
 	private static final Pattern SNAPSHOT_PATTERN = Pattern
 			.compile("^.*[\\.|\\-](BUILD-)?SNAPSHOT.*$");
 
-	private static final String MILESTONE_REGEX = ".*[\\.|\\-]M[0-9]+";
+	private static final String MILESTONE_REGEX = "^.*[\\.|\\-]M[0-9]+.*$";
```
Comment (jonatan-ivanov) [inline releaser-core/src/main/java/releaser/internal/project/ProjectVersion.java:57]: Why do we need `.*` at the end?
I guess, it is how it should be looking at snapshot and rc I just don't know what comes after.

File: releaser-core/src/main/java/releaser/internal/buildsystem/ProjectPomUpdater.java [hunk 2]
```diff
@@ -237,7 +237,7 @@ public FileVisitResult visitFile(Path path, BasicFileAttributes attr) {
 								.stream()
 								.filter(pattern -> IGNORED_SNAPSHOT_LINE_PATTERNS.stream()
 										.noneMatch(line::matches)
-										&& pattern.matcher(line).matches())
+										&& pattern.matcher(line).lookingAt())
```
Comment (jonatan-ivanov) [inline releaser-core/src/main/java/releaser/internal/buildsystem/ProjectPomUpdater.java:240]: Should not the whole thing match?

File: releaser-core/src/main/java/releaser/internal/project/ProjectVersion.java (hunk 1 above)
Comment (spencergibb) [reply releaser-core/src/main/java/releaser/internal/project/ProjectVersion.java:57]: I was just making it look like the other ones.

File: releaser-core/src/main/java/releaser/internal/buildsystem/ProjectPomUpdater.java (hunk 2 above)
Comment (spencergibb) [reply releaser-core/src/main/java/releaser/internal/buildsystem/ProjectPomUpdater.java:240]: I couldn't get it to work any other way. I'm open to ideas.

Comment (jonatan-ivanov) [reply releaser-core/src/main/java/releaser/internal/buildsystem/ProjectPomUpdater.java:240]: ¯\_(ツ)_/¯ :)

**Pull request merged**